src/components/DropDownButton/index.tsx

```diff
@@ -84,17 +85,19 @@ const Toggle = ({
 	handler,
 	label,
 	joined,
+	compact,
 	...props
 }: InternalDropDownButtonProps & {
 	open: boolean;
 	handler: React.MouseEventHandler<HTMLElement>;
 }) => {
+	const shouldCompact = useBreakpoint(compact || [false]);
 	if (joined) {
-		if (label || props.icon) {
+		if (label || shouldCompact) {
 			return (
 				<JoinedButton {...props} onClick={handler}>
+					{!shouldCompact && <Box mr={2}>{label}</Box>}
```

Comment:
Why did we moved this outside the `<Flex>`?
I feel that if we keep that in there there will be zero visual diff.
If we really want to move it, then let's do it in a different isolated PR.